shops/models.py

```diff
@@ -10,7 +10,7 @@ def __str__(self) -> str:
 
 
 class Shop(models.Model):
-    city = models.ForeignKey(City, on_delete=models.PROTECT)
+    city = models.ForeignKey(City, on_delete=models.CASCADE)
```

Comment:
Are you sure about this?

I would think that if there is a shop in the database for a given city, that you shouldn't be able to delete that shop

Instead you could do one of the following:

- Catch the error (see try/except - which we'll cover later) and show an error message
- Disable or not show the delete link if a city has any shops (if city.shop_set.exists() .. or something like that)